### PR TITLE
Fix GCP benchmark workflow error handling [skip ci]

### LINKED ISSUE #406: [BUG] gcs-benchmark workflow error handling, and pre and post ops enhancement
if previous run failed to create cluster, that one would be leave there as "ERROR" state
image

and then following run like https://github.com/NVIDIA/spark-rapids-ml/actions/runs/5971636373/attempts/2 would fail
```
ERROR: (gcloud.dataproc.jobs.submit.pyspark) FAILED_PRECONDITION: Multiple validation errors:
 - No agent on master node(s) reported to be be active
 - Unable to submit job, cluster 'runner-spark-rapids-ml-gpu' is in state ERROR and cannot accept jobs.
```

Some we should do
1. change cluster name to some dynamic one for CI usage (e.g. https://github.com/NVIDIA/spark-rapids-ml/blob/branch-23.08/python/benchmark/dataproc/start_cluster.sh#L45 add support override cluster name)
2. support change COMPUTE_REGION as a workflow param for user to retry when no sufficient resource in default region
3. post-clean up cluster if error. e.g. use https://docs.github.com/en/actions/learn-github-actions/expressions#always to always try delete the cluster (ignore exit code in this step) (or double confirm if max-idle works for cluster in error state)

### PR DESCRIPTION
**Pull request by YanxuanLiu**
fix #406 

To fix the bug
1. change cluster name to dynamic one by adding Github run number as suffix
2. make COMPUTE_REGION as a workflow param for user to retry when no sufficient resource in default region
3. add post-clean up step, always try delete the cluster

### CODE REVIEW

File: python/benchmark/dataproc/run_benchmark.sh [hunk 1]
```diff
@@ -60,7 +60,8 @@ if [[ $? != 0 ]]; then
     exit 1
 fi
 
-cluster_name=${USER}-spark-rapids-ml-${cluster_type}
+cluster_name=${USER}-spark-rapids-ml-${cluster_type}-{GITHUB_RUN_NUMBER}
```
Comment (pxLi) [inline python/benchmark/dataproc/run_benchmark.sh:63]: nit: looks like this script is not for CI only. could try some other way like letting user provide the naming
```
cluster_name=${CLUSTER_NAME:-"${USER}-spark-rapids-ml-${cluster_type}"}
```

File: .github/workflows/gcs-benchmark.yml [hunk 2]
```diff
@@ -17,8 +17,13 @@ name: GCS Benchmark Test
 
 on:
   workflow_dispatch:
+    inputs:
+      computer_region:
+        required: true
+        default: 'us-central1'
+        description: 'Number of Octocats'
```
Comment (pxLi) [inline .github/workflows/gcs-benchmark.yml:24]: nit: seems irrelevant desc

**Review by pxLi — COMMENTED** [review]
LGTM

Please help try some success and fail cases to verify the changes, thanks

**Comment by YanxuanLiu** [comment]
Test builds:

Normal successful run: https://github.com/YanxuanLiu/spark-rapids-ml/actions/runs/6085185863/job/16508682985

Create a cluster manually, and trigger the workflow. It failed as expected, and deleted the cluster successfully at the end of workflow.
https://github.com/YanxuanLiu/spark-rapids-ml/actions/runs/6085897714/job/16511031142

**Review by leewyang — DISMISSED** [review]

**Comment by leewyang** [comment]
Looks like the benchmark is also [running from my fork](https://github.com/leewyang/spark-rapids-ml/actions/runs/6073952952).  Do we need to add code to restrict to main repo somehow?

**Comment by YanxuanLiu** [comment]
> Looks like the benchmark is also [running from my fork](https://github.com/leewyang/spark-rapids-ml/actions/runs/6073952952). Do we need to add code to restrict to main repo somehow?

Yes, if we want to save resource of GSC

**Comment by pxLi** [comment]
> Looks like the benchmark is also [running from my fork](https://github.com/leewyang/spark-rapids-ml/actions/runs/6073952952). Do we need to add code to restrict to main repo somehow?

we should add expression to allow run on prod repo only as defaults
```
if: github.repository == 'NVIDIA/spark-rapids-ml'
```

people could update the value if they want enable in their forks

**Comment by YanxuanLiu** [comment]
> > Looks like the benchmark is also [running from my fork](https://github.com/leewyang/spark-rapids-ml/actions/runs/6073952952). Do we need to add code to restrict to main repo somehow?
> 
> we should add expression to allow run on prod repo only as defaults
> 
> ```
> if: github.repository == 'NVIDIA/spark-rapids-ml'
> ```
> 
> people could update the value if they want enable in their forks

Added and tested https://github.com/YanxuanLiu/spark-rapids-ml/actions/runs/6093671912/job/16533730597

**Review by pxLi — APPROVED** [review]
Cool! please trigger a test run after merge

**Comment by pxLi** [comment]
build

**Pull request merged**